src/main/java/com/icthh/xm/ms/entity/service/CalendarService.java

```diff
@@ -51,9 +54,24 @@ public Calendar save(Calendar calendar) {
                                                               Calendar::setStartDate,
                                                               Calendar::getStartDate);
         calendar.setXmEntity(xmEntityRepository.getOne(calendar.getXmEntity().getId()));
+        assertReadOnlyCalendar(calendar);
         return calendarRepository.save(calendar);
     }
 
+    public void assertReadOnlyCalendar(Calendar calendar) {
+        if (calendar == null || calendar.getId() == null) {
+            return;
+        }
+
+        specService
+            .findCalendar(calendar.getXmEntity().getTypeKey(), calendar.getTypeKey())
+            .map(CalendarSpec::getReadonly)
+            .filter(isReadonly -> isReadonly)
```

Comment:
Function::identity

Reply:
https://stackoverflow.com/questions/43785804/why-cant-i-replace-t-t-with-function-identity-in-the-argument-to-optiona